cmd/grpcurl/grpcurl.go

```diff
@@ -426,6 +432,16 @@ func main() {
 		} else if *authority != "" {
 			opts = append(opts, grpc.WithAuthority(*authority))
 		}
+		if *userAgent != "" {
```

Comment:
I think you need to remove this outer check, since you check for this inside (and you don't need this flag to short-circuit the `grpcurl/...` part of the UA.

Reply:
🤦‍♂️ good catch!